tidy3d/components/data/monitor_data.py

```diff
@@ -320,6 +350,22 @@ def flux(self) -> FluxDataArray:
         d_area = self._diff_area
         return FluxDataArray((self.poynting * d_area).sum(dim=d_area.dims))
 
+    @cached_property
+    def mode_area(self) -> FreqModeDataArray:
+        """Effective mode area corresponding to a 2D monitor
+
+        Effective mode area is calculated as: (∫|E|²dA)² / (∫|E|⁴dA)
+        """
+        fields = self._centered_fields
```

Comment:
Sorry, I see why. I think in the interest of avoiding repeating code, we should reorganize things though. We can have `_tangential_centered_fields` take `_centered_fields` and pick out the tangential components.

Reply:
Because it also needs the longitudinal components.

Reply:
Yes re sign inversion; re grid correction, you should apply the `primal` grid correction to the normal H field, and the `dual` grid correction to the normal E field because that's where these components live along that axis.

Reply:
Thanks, I'll fix the corrections. This sign inversion for the purpose of inverting the propagation direction direction might be a problem in the future, though. When fully anisotropic magnetic materials are implemented, it is possible to build unidirectional waveguide modes, so the sign cannot be not freely swapped. I guess this is a problem for later, but we should remember it.

Reply:
The sign inversion here is not for the purpose of propagation direction inversion, but rather because we are working in (x, z, y) reference frame, which is left-handed. Your general comment is definitely valid for the case of computing expansion coefficients though, as there we do not re-compute the backward propagating ones currently, but just use the time-reversed version of the forward propagating ones. Also, this conversion to and from a left-handed frame when the normal axis is y may need something extra in that case, namely some of the off-diagonal eps components will have to get a - sign?

I guess it would be great to improve this handling in general... We have a number of places in the code where we check `if normal_ind == 1`. I kinda regret not working in (z, x, y) reference frame from the start, but changing to that now is its own can of worms.